android/tangram/jni/platform_android.cpp

```diff
@@ -266,5 +276,16 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, tid, priority);
 }
 
+void featureSelectionCallback(JNIEnv* jniEnv, const std::vector<std::shared_ptr<Tangram::Properties>>& items) {
+
+    jlong jresult = 0;
+    jclass propertiesClass = jniEnv->FindClass("com/mapzen/tangram/Properties");
+
+    jresult = reinterpret_cast<jlong>(new std::shared_ptr<Tangram::Properties >(items[0]));
```

Comment:
Priority is on getting this merged, and we can extend behavior later. I
think either will work initially here.


On Monday, October 5, 2015, Hannes Janetzek <notifications@github.com>
wrote:

> In android/tangram/jni/platform_android.cpp
> <https://github.com/tangrams/tangram-es/pull/292#discussion_r41206837>:
>
> > @@ -266,5 +276,16 @@ void setCurrentThreadPriority(int priority) {
> >      setpriority(PRIO_PROCESS, tid, priority);
> >  }
> >
> > +void featureSelectionCallback(JNIEnv* jniEnv, const std::vector<std::shared_ptr<Tangram::Properties>>& items) {
> > +
> > +    jlong jresult = 0;
> > +    jclass propertiesClass = jniEnv->FindClass("com/mapzen/tangram/Properties");
> > +
> > +    jresult = reinterpret_cast<jlong>(new std::shared_ptr<Tangram::Properties >(items[0]));
>
> Yes, I also think that there are cases where it is good to return a set of
> items near the touch point and let the client decide if it changes its
> previous selection (or whatever). But I think we can postpone this unless
> eraser-map plans to use it now
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/292/files#r41206837>.
>